components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/handler/UserEmailVerificationHandler.java

```diff
@@ -421,8 +422,7 @@ private String getPendingVerificationEmailValue(UserStoreManager userStoreManage
 
         Map<String, String> verificationPendingEmailClaimMap;
         try {
-            verificationPendingEmailClaimMap = userStoreManager.getUserClaimValues(IdentityUtil.addDomainToName(
-                    user.getUserName(), user.getUserStoreDomain()), new String[]{
+            verificationPendingEmailClaimMap = userStoreManager.getUserClaimValues(user.getUserName(), new String[]{
```

Comment:
Check whether username contains the domain or whether its needed?

Reply:
`user.getUserName()` does not contain domain. To get the domain `user.getTenantDomain` method needs to be used.